lib/workers/global/autodiscover.ts

```diff
@@ -30,7 +30,17 @@ export async function autodiscoverRepositories(
     return config;
   }
   if (config.autodiscoverFilter) {
-    discovered = discovered.filter(minimatch.filter(config.autodiscoverFilter));
+    const matched: string[] = [];
```

Comment:
```suggestion
    const matched = new Set<string>();
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set